finesse/gui/docs_view.py

```diff
@@ -0,0 +1,62 @@
+"""Code for FINESSE's documentation viewer window."""
+from importlib import resources
+from pathlib import Path
+
+from PySide6.QtCore import QUrl
+from PySide6.QtGui import QAction
+from PySide6.QtWebEngineWidgets import QWebEngineView
+from PySide6.QtWidgets import QMainWindow, QToolBar, QVBoxLayout, QWidget
+
+
+class DocsViewer(QAction):
+    """A window for viewing documentation."""
+
+    def __init__(self, parent) -> None:
+        """Create a push button for opening documentation in a new window."""
+        super().__init__("Help", parent)
+        self.triggered.connect(self.show_docs)
+
+    def show_docs(self) -> None:
+        """Create the documentation in a new window."""
+        self.docs_window = self.create_docs_window()
+        self.docs_window.show()
+
+    def create_docs_window(self) -> QMainWindow:
+        """Create a new window displaying the documentation.
+
+        Returns:
+            A main window showing the html documentation.
+        """
+        docs_window = QMainWindow()
+        docs_window.setWindowTitle("FINESSE User Guide")
+
+        toolbar = QToolBar()
+        toolbar.setMovable(False)
+        home_btn = toolbar.addAction("Home")
+        back_btn = toolbar.addAction("Back")
+        forward_btn = toolbar.addAction("Forward")
+        docs_window.addToolBar(toolbar)
+
+        docs_path = resources.files("docs")
+        self.docs_home = Path(str(docs_path.joinpath("user_guide.html")))
+        if not self.docs_home.exists():
+            raise FileNotFoundError("User guide not found.")
+        self.browser = QWebEngineView()
+        self._open_homepage()
+        home_btn.triggered.connect(self._open_homepage)
+        back_btn.triggered.connect(self.browser.back)
+        forward_btn.triggered.connect(self.browser.forward)
+
+        layout = QVBoxLayout()
+        layout.addWidget(self.browser)
+
+        central = QWidget()
+        central.setLayout(layout)
+
+        docs_window.setCentralWidget(central)
+
+        return docs_window
+
+    def _open_homepage(self):
+        """Go to documentation home page."""
+        self.browser.load(QUrl(f"file://{self.docs_home}"))
```

Comment:
Hmm, I'm slightly concerned that special chars in the path might not be properly escaped here.

The following works for me:

```suggestion
        self.browser.load(self.docs_home.as_uri())
```

Reply:
Ah yeah, that's what I was attempting to do [here](https://github.com/ImperialCollegeLondon/FINESSE/pull/412#discussion_r1392265956), I needed to remove the QUrl 😔